core/eolearn/core/utils/parsing.py

```diff
@@ -81,25 +98,35 @@ class FeatureParser:
     - For `get_renamed_features` a list of triples `(feature_type, old_name, new_name)`.
     """
 
-    def __init__(self, features: Union[dict, Sequence], allowed_feature_types: Optional[Iterable[FeatureType]] = None):
+    def __init__(
+        self,
+        features: Union[SingleFeatureSpec, MultiFeatureSpec],
```

Comment:
I know it might be too late to make such a change now, but the name `features` suggests plural, so perhaps we could drop this convenience of auto recognizing if the argument is a single feature or a list of features. The user would just always provide a list. I believe that would make it simpler, but I'm aware that making such a change for an open-source package might not be as simple.

Anyway, feel free to ignore, just pointing it out.

Reply:
I tried doing it in `eo-learn 1.0` but it would break a lot of the tasks. Perhaps by introducing gradual changes we can slowly get the parser to a manageable level.

Reply:
I agree it would introduce breaking changes, that was expected, but I still think it might be worth it to do it over the span of a few releases, first adding a deprecation warning and then removing it. Perhaps a good first task?

Reply:
Feel free to resolve if this is outdated or if you don't plan on making any further changes related to this.